pyccel/ast/builtin_methods/set_methods.py

```diff
@@ -123,3 +123,31 @@ def __init__(self, set_variable):
         self._rank = set_variable._rank
         self._class_type = set_variable._class_type
         super().__init__(set_variable)
+
+
+class SetPop(SetMethod):
+    """
+    Represents a call to the .pop() method.
+
+    The pop() method pops an element from the set. 
+    It does not take any arguments but returns the popped 
+    element. It raises an error if the set is empty.
+    The class does not raise an error as it assumes that the
+    user code is valid.
+
+    Parameters
+    ----------
+    set_variable : TypedAstNode
+        The name of the set.
+    """
+    __slots__ = ('_dtype', '_precision', '_class_type')
+    _rank = 0
+    _order = None
+    _shape = None
+    name = 'pop'
+
+    def __init__(self, set_variable):
+        self._dtype = set_variable.dtype
+        self._precision = set_variable.precision
+        self._class_type = set_variable.dtype
```

Comment:
Please update this section now that #1756 has been merged

Reply:
Fixed : cfe3c6f